src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1754,12 +1790,19 @@ public static ErlangIncludeString getIncludeStringSafe(@NotNull ErlangIncludeLib
 
   @NotNull
   public static ErlangFunctionReferenceImpl<ErlangQAtom> createFunctionReference(@NotNull ErlangQAtom atom) {
-    PsiElement parent = atom.getParent();
-    ErlangMaxExpression moduleExpression = PsiTreeUtil.getPrevSiblingOfType(parent, ErlangMaxExpression.class);
-    ErlangListExpression list = PsiTreeUtil.getNextSiblingOfType(parent, ErlangListExpression.class);
-    ErlangQAtom module = moduleExpression == null ? null : moduleExpression.getQAtom();
-    int arity = list == null ? -1 : list.getExpressionList().size();
-    return new ErlangFunctionReferenceImpl<ErlangQAtom>(atom, module, arity);
+    ErlangFunctionCallExpression call = PsiTreeUtil.getParentOfType(atom, ErlangFunctionCallExpression.class);
+    if (call != null) {
+      int position = getPositionInFunctionCall(call, atom);
+      List<ErlangExpression> expressions = call.getArgumentList().getExpressionList();
+      if (0 < position && position + 1 < expressions.size()) {
```

Comment:
Please, do your comparisons uniformly. Here you first have your checked value on the right hand side, then - on the left. It's hard to read - choose one order and stick to it everywhere.